### PR TITLE
Small code improvements in Parser + test

### PR DESCRIPTION
**Pull request by hyperxor**
I have made small improvements in `Parser` class implementation and one test

### CODE REVIEW

File: test/parser_test.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+TEST(ParserTest, Empty) {
+    Parser parser;
+
+    EXPECT_FALSE(parser);
+
+    MockEventHandler handler;
+    EXPECT_FALSE(parser.HandleNextDocument(handler));
+    
+    std::ostringstream oss;
+    parser.PrintTokens(oss);
```
Comment (jbeder) [inline test/parser_test.cpp]: Maybe instead of checking the token list, verify that there are no interactions with the MockEventHandler?

Comment (hyperxor) [reply test/parser_test.cpp]: @jbeder Yes, good proposal! I will do it. I have some doubts about doing it _instead_. I would add it as an addition to current checks. What do you think?

Comment (jbeder) [reply test/parser_test.cpp]: The reason I'd suggest *instead* is that Parser::PrintTokens is a debugging function (it's actually unused, and probably should just be deleted...), so it's not good to rely on its output in a test.

Comment (hyperxor) [reply test/parser_test.cpp]: @jbeder I have done it

File: test/parser_test.cpp [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+using YAML::MockEventHandler;
+using ::testing::StrictMock;
+
+TEST(ParserTest, Empty) {
+    Parser parser;
+
+    EXPECT_FALSE(parser);
+
+    StrictMock<MockEventHandler> handler;
+    parser.HandleNextDocument(handler);
```
Comment (jbeder) [inline test/parser_test.cpp]: Why not keep the expect false?

File: test/parser_test.cpp [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    
+    std::ostringstream oss;
```
Comment (jbeder) [inline test/parser_test.cpp]: Remove this, as discussed?

Comment (hyperxor) [reply test/parser_test.cpp]: @jbeder sorry, did wrong change, now everything looks as discussed

File: test/parser_test.cpp (hunk 2 above)
Comment (hyperxor) [reply test/parser_test.cpp]: @jbeder I did wrong commit, sorry, now everything is good

**Pull request merged**